Dockerfile-local

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG CP_VERSION
+
+FROM confluentinc/cp-server-connect-base:$CP_VERSION
+
+ARG CONNECTOR_VERSION=5.5.0-SNAPSHOT
+
+ENV CONNECT_PLUGIN_PATH: "/usr/share/java,/connect-plugins,/usr/share/confluent-hub-components"
+
+# Install Confluent Replicator connector
+COPY confluentinc-kafka-connect-replicator-${CONNECTOR_VERSION}.zip /tmp/confluentinc-kafka-connect-replicator-${CONNECTOR_VERSION}.zip
+RUN confluent-hub install --no-prompt /tmp/confluentinc-kafka-connect-replicator-${CONNECTOR_VERSION}.zip
+
+# Install Elasticsearch connector
+COPY confluentinc-kafka-connect-elasticsearch-${CONNECTOR_VERSION}.zip /tmp/confluentinc-kafka-connect-elasticsearch-${CONNECTOR_VERSION}.zip
+RUN confluent-hub install --no-prompt /tmp/confluentinc-kafka-connect-elasticsearch-${CONNECTOR_VERSION}.zip
```

Comment:
Is downloading, or installing, or both - connectors - generally expensive?  If it could be avoided for every container start, and just happen once, would that be enough?

Local image builds are OK when needed or for production, but if this was avoidable for dev in the future with a single-install and the use of persistent volumes, that would be my preference, provided a single download and/or install isn't too expensive.

Would these two versions of the Dockerfile continue to diverge, or could the differences be moved to variables and an install script driven by a single file?

Reply:
@javabrett IIUC the question, Docker is already using a cache so the build of the connect image happens only once.

```
[10:36:49] ~/git/cp-demo(DEVX-1575): docker images | grep local
localbuild/connect                                                                       5.5.x-latest-5.5.0-SNAPSHOT   f1bed9a6f77f        2 days ago          1.46GB
localbuild/connect                                                                       5.5.x-latest-5.4.1            6a0caac9f605        2 days ago          1.39GB
```

Notice it doesn't pull the connectors are again

```
Sending build context to Docker daemon  159.9MB
Step 1/6 : ARG CP_VERSION
Step 2/6 : FROM confluentinc/cp-server-connect-base:$CP_VERSION
 ---> 4d09ff44314b
Step 3/6 : ARG CONNECTOR_VERSION=5.4.1
 ---> Using cache
 ---> 877095b2d93b
Step 4/6 : ENV CONNECT_PLUGIN_PATH: "/usr/share/java,/connect-plugins,/usr/share/confluent-hub-components"
 ---> Using cache
 ---> f9b41dd91612
Step 5/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-replicator:${CONNECTOR_VERSION}
 ---> Using cache
 ---> db445b5cafea
Step 6/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-elasticsearch:${CONNECTOR_VERSION}
 ---> Using cache
 ---> 6a0caac9f605
Successfully built 6a0caac9f605
Successfully tagged localbuild/connect:5.5.x-latest-5.4.1
```

As for the second question about a single Dockerfile: I originally started as a single Dockerfile and uses ARGs but got stuck and so split it into two (I think the issue was having a `COPY` inside a `RUN`).  The current method of two files is also consistent with https://github.com/confluentinc/kafka-connect-datagen/tree/master.  Feel free to file a jira for a separate PR that we can consider for future, but it would require more investigation